azure/scope/identity.go

```diff
@@ -159,6 +160,14 @@ func (p *AzureCredentialsProvider) GetTokenCredential(ctx context.Context, resou
 		}
 		cred, authErr = p.cache.GetOrStoreManagedIdentity(&options)
 
+	case infrav1.UserAssignedIdentityCredential:
+		cloudType := parseCloudType(p.Identity.Spec.UserAssignedIdentityCredentialsCloudType)
+		clientOptions := azcore.ClientOptions{
+			TracingProvider: tracingProvider,
+			Cloud:           cloudType,
+		}
+		cred, authErr = p.cache.GetOrStoreUserAssignedManagedIdentityCredentials(context.Background(), p.Identity.Spec.UserAssignedIdentityCredentialsPath, clientOptions, &log)
```

Comment:
Non-blocking comment: 
Could we please add a Note here highlighting the reason for choosing `context.Background()` instead of `context.WithValues` or other choices ? A reminder for us to be able to fix the context deadline exceeded logs we've have been seeing..

Reply:
I updated and added a comment